ow_sim_tests/scripts/test_arm_consistency.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+
+# The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+# Research and Simulation can be found in README.md in the root directory of
+# this repository.
+
+import rospy
+import actionlib
+import pandas as pd
+import tf2_ros
+import roslib
+import unittest
+import owl_msgs.msg
+import numpy as np
+import random
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
+from geometry_msgs.msg import Pose, Quaternion
+from action_testing import *
+from fastdtw import fastdtw
+from scipy.spatial.distance import euclidean
+
+PKG = 'ow_sim_tests'
+TEST_NAME = 'arm_consistency'
+roslib.load_manifest(PKG)
+
+def _dtw_helper(trajectories, runs):
+  baseline_index = 0
+
+  threshold = 10
+
+  while not rospy.is_shutdown():
+    baseline_trajectory = trajectories[baseline_index]
+    outlier_indices = []
+    for i, traj in enumerate(trajectories[1:]):
+      distance, _ = fastdtw(baseline_trajectory, traj, dist = euclidean)
+      if distance > threshold:
+        outlier_indices.append(i+1)
+    if len(outlier_indices) >= (runs / 2):
+      baseline_index += 1
+    else:
+      break
+      
+  return outlier_indices, threshold
+
+def _plot_helper(trajectories):
+  fig = plt.figure()
+  ax = fig.add_subplot(111, projection='3d')
+  
+  for i, traj in enumerate(trajectories):
+    traj = np.array(traj)
+    xs, ys, zs = traj[:, 0], traj[:, 1], traj[:, 2]
+    ax.plot(xs, ys, zs)
+
+  ax.set_xlabel('X')
+  ax.set_ylabel('Y')
+  ax.set_zlabel('Z')
+  plt.title('3D Trajectories')
+  plt.legend()
+  # TODO change the saving location in your local directory
+  fig.savefig('/home/honghao/Documents/3D_trajectories_RRTConnect_testing.png')
```

Comment:
We can't have a user's directory hardcoded here.  I'll suggest a portable pathname such as `~/.ros/...` or `~/Desktop/...`.   There are several more of these in this file.